src/markdown.cpp

```diff
@@ -1846,18 +1843,22 @@ static bool isHRuler(const char *data,int size)
 }
 
 std::unordered_map<std::string, int> id_count;
-QCString githubHtmlIdentifier(std::string str)
+QCString githubHtmlIdentifier(const std::string &str)
 {
-  // Replace space with '-'
-  std::replace(str.begin(), str.end(), ' ', '-');
+  static std::string_view germanSharpS = "ß";
```

Comment:
There are thousands of non-ascii characters, why is only this one getting a special treatment? You should assume the string has UTF-8 encoding, see utf8.h for helper functions.

Reply:
For the ß, I know GitHub leaves it in the id, see https://github.com/tim-gromeyer/stackoverflow/blob/tests/README.md

Reply:
Please also add in comment a list of rules to show how characters are replaced.